### PR TITLE
Add resources and ledger_resources to API

### LINKED ISSUE #9: CRUD API Endpoints for ledger resources (2)
We want the ability to create/update/read/delete a ledger resource in Ledger Sync without performing an actual sync. This can be useful when we know an entity already exists in the ledger so we don't want to overwrite its attributes/perform an operation

### PR DESCRIPTION
**Pull request by ryanwjackson**
Fix #9

### CODE REVIEW

File: app/models/concerns/externally_identifiable.rb [hunk 1]
```diff
@@ -4,6 +4,19 @@ module ExternallyIdentifiable
   extend ActiveSupport::Concern
 
   module ClassMethods
+    def efind(use_id)
```
Comment (mattgmarcus) [inline app/models/concerns/externally_identifiable.rb]: When I first saw this method, I thought it was a typo. The name's a bit weird. Maybe `find_by_best_id` or `find_by_id_or_external_id` (verbose, I know)

File: app/models/concerns/externally_identifiable.rb [hunk 2]
```diff
@@ -4,6 +4,19 @@ module ExternallyIdentifiable
...
+      find_by(id: use_id) || find_by(external_id: use_id)
+    end
+
+    def efind!(use_id, api: false)
+      ret = efind(use_id)
+      return ret if ret.present?
+
+      raise NoSuchRecordError.new(self::API_OBJECT, external_id: true) if api
+
+      raise ActiveRecord::RecordNotFound, "#{name} could not be found with id or external_id: #{use_id}"
```
Comment (mattgmarcus) [inline app/models/concerns/externally_identifiable.rb]: Is `name` defined somewhere else?

File: app/serializers/resource_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# == Schema Information
+#
+# Table name: users
```
Comment (mattgmarcus) [inline app/serializers/resource_serializer.rb]: Typo - `resources` for table name?

Comment (mattgmarcus) [reply app/serializers/resource_serializer.rb]: I think it's wrong below too

File: spec/lib/forms/ledger_resources/destroy_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  it { expect_valid }
+  it { expect(result).to be_success }
+  it { expect(value).to be_a(LedgerResource) }
+
+  describe '#ledger_resource' do
+    it { expect_error_with_missing_attribute(:ledger_resource) }
+    xit { expect_error_with_attribute_value(:ledger_resource, LEDGER_RESOURCE_BAD_VALUE, message: nil) } # :message is optional
+    xit { expect_valid_with_attribute_value(:ledger_resource, LEDGER_RESOURCE_GOOD_VALUE) }
```
Comment (mattgmarcus) [inline spec/lib/forms/ledger_resources/destroy_spec.rb]: Gonna keep these pending?

**Comment by mattgmarcus** [comment]
@ryanwjackson I realize you didn't do the Delete operation for either one of these resources. Was there a reason for that?

File: app/models/concerns/externally_identifiable.rb (hunk 2 above)
Comment (ryanwjackson) [reply app/models/concerns/externally_identifiable.rb]: This is a concern included in a class, so `name` is the name of the class, in this case our models.  So it's something like `Customer could not be found...`

**Comment by ryanwjackson** [comment]
@mattgmarcus What do you mean?  There are destroy forms and methods on the controllers?

**Comment by mattgmarcus** [comment]
@ryanwjackson my bad. I didn't see it documented in README.md but I see it now.

**Pull request merged**